### PR TITLE
feat(launch-agent): Configuration changes

### PR DESCRIPTION
**Pull request by wandb-zacharyblasczyk**
* Launch-config parameterized
* Deployment takes `.Values.launchConfig.base_url`
* `annotations.checksum/config` on deployment to force replacement if CM changes
* `.Values.namespace` is now configured
* Fix launch `ct lint --config ct.yaml`
* update launch documentation

### CODE REVIEW

File: charts/launch-agent/values.yaml [hunk 1]
```diff
@@ -11,8 +11,34 @@ agent:
...
+  environment:
+    type: local #aws|gcp
+
+  # Container registry config.
+  registry:
+    type: local #ecr|gcr
+
+  # Container build config.
+  builder:
+    type: docker #kaniko|noop
```
Comment (bcsherma) [inline charts/launch-agent/values.yaml]: the build should default to noop, I don't think anyone will use the docker builder within k8s

File: charts/launch-agent/templates/rbac.yaml [hunk 2]
```diff
@@ -1,17 +1,19 @@
+---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: wandb-launch-serviceaccount
-  namespace: wandb
+  namespace: {{ .Values.namespace }}
```
Comment (bcsherma) [inline charts/launch-agent/templates/rbac.yaml:6]: I'm not we can support installing in another namespace, I believe that the `wandb` namespace is hard coded in a few places in the sdk

**Review by bcsherma — COMMENTED** [review]
We'll need to make some SDK changes to support installing the agent in a namespace other than "wandb". Can you also update the README.md in charts/launch-agent/ with these value changes?

File: charts/launch-agent/values.yaml (hunk 1 above)
Comment (wandb-zacharyblasczyk) [reply charts/launch-agent/values.yaml]: we can default to noop but then could you submit a pr to update the docs:
https://docs.wandb.ai/guides/launch/run-agent#agent-configuration

```
The environment, registry, and builder keys are optional. By default, the agent will use no cloud environment, a local docker registry, and a docker builder.
```

File: charts/launch-agent/README.md [hunk 3]
```diff
@@ -4,33 +4,42 @@ This chart deploys the W&B Launch Agent to your Kubernetes cluster.
...
+| `agent.resources`             | object          | No       | Limit to 1 CPU, 1Gi RAM           | Pod spec resources block for the agent.                                                                                                          |
+| `namespace`                   | string          | No       | `wandb`                           | The namespace to deploy the agent into.                                                                                                          |
+| `additionalTargetNamespaces`  | list(string)    | No       | [`wandb`,`default`]               | The namespace to deploy the agent into.                                                                                                          |
+| `launchConfig`                | mutiline string | **Yes**  | `https://api.wandb.ai`            | URL of your W&B server.                                                                                                                          |
```
Comment (KyleGoyette) [inline charts/launch-agent/README.md:41]: I think the vriable here should be baseUrl? And we need another line for launchConfig?

File: charts/launch-agent/ci/basic-values.yaml [hunk 4]
```diff
@@ -1,4 +1,25 @@
 agent:
-  apiKey: "<an-api-key>"
+  apiKey: local-...
```
Comment (KyleGoyette) [inline charts/launch-agent/ci/basic-values.yaml:2]: Should this give an example with local?

File: charts/launch-agent/templates/namespace.yaml [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
-  name: wandb
+  name: {{ .Values.namespace }}
   labels:
     pod-security.kubernetes.io/enforce: baseline
     pod-security.kubernetes.io/enforce-version: latest
     pod-security.kubernetes.io/warn: baseline
-    pod-security.kubernetes.io/warn-version: latest
\ No newline at end of file
+    pod-security.kubernetes.io/warn-version: latest
+...
```
Comment (KyleGoyette) [inline charts/launch-agent/templates/namespace.yaml:11]: what does ... mean here?

Comment (KyleGoyette) [reply charts/launch-agent/templates/namespace.yaml:11]: Does this need an {{- end}}?

File: charts/launch-agent/templates/volcano.yaml [hunk 6]
```diff
@@ -1,5 +1,5 @@
-# condition this whole file out if the user doesn't want to install the admission controller
 {{ if .Values.volcano }}
+# condition this whole file out if the user doesn't want to install the admission controller
```
Comment (KyleGoyette) [inline charts/launch-agent/templates/volcano.yaml:2]: We've done that now yeah?

**Review by KyleGoyette — COMMENTED** [review]
I have less context than Ben here, but left a few comments

File: charts/launch-agent/README.md [hunk 7]
```diff
@@ -4,33 +4,42 @@ This chart deploys the W&B Launch Agent to your Kubernetes cluster.
...
+| `agent.imagePullPolicy`       | string          | No       | `Always`                          | Pull policy for the agent container image.                                                                                                       |
+| `agent.resources`             | object          | No       | Limit to 1 CPU, 1Gi RAM           | Pod spec resources block for the agent.                                                                                                          |
+| `namespace`                   | string          | No       | `wandb`                           | The namespace to deploy the agent into.                                                                                                          |
+| `additionalTargetNamespaces`  | list(string)    | No       | [`wandb`,`default`]               | The namespace to deploy the agent into.                                                                                                          |
```
[the remark this answers is not in the document]
Comment (wandb-zacharyblasczyk) [reply charts/launch-agent/README.md:40]: Updated.

File: charts/launch-agent/README.md (hunk 3 above)
Comment (wandb-zacharyblasczyk) [reply charts/launch-agent/README.md:41]: Updated

File: charts/launch-agent/ci/basic-values.yaml (hunk 4 above)
Comment (wandb-zacharyblasczyk) [reply charts/launch-agent/ci/basic-values.yaml:2]: Updated.

File: charts/launch-agent/templates/namespace.yaml (hunk 5 above)
Comment (wandb-zacharyblasczyk) [reply charts/launch-agent/templates/namespace.yaml:11]: ... means it is the end of that yaml section.

Comment (wandb-zacharyblasczyk) [reply charts/launch-agent/templates/namespace.yaml:11]: @bcsherma, should we wrap this?

File: charts/launch-agent/templates/volcano.yaml (hunk 6 above)
Comment (bcsherma) [reply charts/launch-agent/templates/volcano.yaml:2]: yeah this is done

File: charts/launch-agent/templates/namespace.yaml (hunk 5 above)
Comment (bcsherma) [reply charts/launch-agent/templates/namespace.yaml:11]: Not sure what you mean exactly?

**Review by bcsherma — APPROVED** [review]
Looks great ❤️‍🔥

**Pull request merged**